rest_framework/serializers.py

```diff
@@ -268,7 +269,7 @@ def _get_declared_fields(cls, bases, attrs):
             if hasattr(base, '_declared_fields'):
                 fields = list(base._declared_fields.items()) + fields
 
-        return OrderedDict(fields)
+        return attrs.get('_dict_class', OrderedDict)(fields)
```

Comment:
What's the harm in using the alternative dictionary class for the fields themselves?
The motiviation behind this pull request is to use the C version of the ordered dictionary written in Cython or just the built in dictionary and the serializers works with both.